OneSchedule.Domain/NotificationScheduler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.Extensions.Configuration;
+using OneSchedule.Domain.Abstractions;
+using OneSchedule.Entities;
+using Quartz;
+using Quartz.Impl;
+using Quartz.Lambda;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Telegram.Bot;
+
+namespace OneSchedule.Domain
+{
+    public class NotificationScheduler : INotificationScheduler
+    {
+        private readonly ITelegramBotClient _bot;
+        private readonly IConfiguration _configuration;
+
+        public NotificationScheduler(ITelegramBotClient bot, IConfiguration configuration)
+        {
+            _bot = bot;
+            _configuration = configuration;
+        }
+
+        public async Task ScheduleNotifications(EventEntity eventEntity)
+        {
+            IScheduler scheduler = await SchedulerRepository.Instance.Lookup(_configuration.GetSection("Quartz").GetSection("quartz.scheduler.instanceName").Value);
```

Comment:
Is there any way not to inject the whole configuration and pass only required data? Also you don't need to use two calls of GetSection() method, you can specify required section in one call using nesting

Reply:
Usedone call approach

Reply:
*Used